ntp-udp/src/raw_socket/recv_message.rs

```diff
@@ -1,21 +1,111 @@
-use std::os::unix::prelude::AsRawFd;
+use std::{io::IoSliceMut, marker::PhantomData, net::SocketAddr, os::unix::prelude::AsRawFd};
+
+use tracing::warn;
+
+use crate::interface_name::sockaddr_storage_to_socket_addr;
 
 use super::cerr;
 
-/// Receive a message on a socket (retry if interrupted)
-pub(crate) fn receive_message(
+pub(crate) enum MessageQueue {
```

Comment:
This commit has too much going on for me to review efficiently. I'd recommend cutting this up in smaller pieces which should make it easier to verify each change in isolation.

Reply:
Split this commit into multiple